containers/src/Kokkos_DynRankView.hpp

```diff
@@ -580,9 +580,37 @@ class DynRankView : private View<DataType*******, Properties...> {
              index_type i6 = 0) const {
     return view_type::operator()(i0, i1, i2, i3, i4, i5, i6);
   }
+
+// This is an accomodation for Phalanx, who use the operator[] to access all
+// memory
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
   KOKKOS_FUNCTION reference_type operator[](index_type i0) const {
+    if constexpr (std::is_same<typename drvtraits::value_type,
+                               typename drvtraits::scalar_array_type>::value) {
+      return view_type::data()[i0];
+    } else {
+      const size_t dim_scalar = view_type::impl_map().dimension_scalar();
+      const size_t bytes      = view_type::span() / dim_scalar;
+
+      using tmp_view_type =
+          Kokkos::View<DataType*, typename traits::array_layout,
+                       typename traits::device_type,
+                       Kokkos::MemoryTraits<traits::memory_traits::impl_value |
+                                            unsigned(Kokkos::Unmanaged)>>;
```

Comment:
I can't believe we did not mandate the underlying integral type of `MemoryTraitsFlags` is not `unsigned` in the first place and we cast everywhere...

Reply:
Yes sorry I am just ruminating.  Didn't mean to suggest there is anything to fix here